tests/test_nestml_printer.py

```diff
@@ -47,73 +48,73 @@ def setUp(self):
         Logger.init_logger(LoggingLevel.INFO)
 
     def test_block_with_variables_with_comments(self):
-        block = "# pre1\n" \
-                "state: # in\n" \
-                "    # real pre\n" \
-                "    # real pre2\n" \
-                "    r real = 0\n"
-        model = ModelParser.parse_block_with_variables(block)
+        block = "    # pre1\n" \
+                "    state: # in\n" \
+                "        # real pre\n" \
+                "        # real pre2\n" \
+                "        r real = 0\n"
+        model = ModelParser.parse_model_body(block)
```

Comment:
Would this be any different if parsed as `block_with_variables`?

Reply:
That wouldn't work, because it cannot account for the pre comment (which leaves behind stray "\n" characters as the comments go into the COMMENTS channel during parsing.